docs: add helm images documentaiton (#930)
okteto/docs: src/content/self-hosted/helm-configuration.mdx

  slackWebhook:
```

> ### backend

Reviewer: `backend` is not a component, not sure where is the best place to document this.
We already have a `globals.jobs` section for settings affecting all the jobs. Maybe moving it to `globals.backend` for settings affecting all the backend services?
Ideally, we should have the `image` section on each backend service, but that's out of the scope of this work.
@ifbyol any ideas?

Reviewer: Another option is not to document all these image fields. We only expect people to modify:
- `globals.registry`
- `cli.image`
- `installer.runner`
All other image fields shouldn't be configured by our users

Reviewer 2: It is a good question. `global.backend` could be misleading as BuildKit or Registry are not affected by that, and it is hard to know why some components would be included as `backend` and not others.

I would go with no documenting that section for now.

Reviewer: SGTM

The backend service in Okteto handles the core functionalities of your application.

> - `image`: Specifies the image settings for your backend service.

Reviewer: ```suggestion
- `image`: Specifies the image settings for your backend services.
```

>   - `registry`: The registry where the backend image is hosted. Defaults to `` (You can change all the registries by setting `globals.registry` to your registry).

Author: Where should i put `(You can change all the registries by setting `globals.registry` to your registry)` message?

>   - `registry`: The registry where the backend image is hosted. Defaults to `` (You can change all the registries by setting `globals.registry` to your registry).

Reviewer: ```suggestion
  - `registry`: The registry where the backend image is hosted. Defaults to the value of `globals.registry`.
```

>   - `repository`: The repository path for the backend image. Defaults to `okteto/backend` if not specified.

Reviewer: ```suggestion
  - `repository`: The repository path for the backend image. Defaults to `okteto/backend`.
```

>   - `tag`: The image tag to deploy. Defaults to `{variables.chartVersion}` if not specified.

Reviewer: ```suggestion
  - `tag`: The image tag for the backend image. Defaults to `{variables.chartVersion}`.
```

```yaml
backend:
  image:

>     registry: myregistry.com

Reviewer: let's use as a sample docker hub, okteto/backend and the chart tag

- `hpa.min`: Minimum number of buildkit pods to keep running.
- `hpa.max`: Maximum number of buildkit pods to scale to.
- `hpa.cpu`: The amount of CPU utilization that will cause the HPA to scale the buildkit pods.
- `image`: Specifies the image settings for your buildkit service.
  - `registry`: The registry where the buildkit image is hosted. Defaults to `` (You can change all the registries by setting `globals.registry` to your registry).
  - `repository`: The repository path for the buildkit image. Defaults to `okteto/buildkit` if not specified.

>   - `tag`: The image tag to deploy. Defaults to `0.17.2-okteto1` if not specified.

Author: I don't like to set the default value here, it would imply changing it almost every release

Reviewer: My understanding is that, as part of this work, we were tagging buildkit and registry with the chart release, didn't we do that?

- `hpa.min`: Minimum number of buildkit pods to keep running.
- `hpa.max`: Maximum number of buildkit pods to scale to.
- `hpa.cpu`: The amount of CPU utilization that will cause the HPA to scale the buildkit pods.

> - `image`: Specifies the image settings for your buildkit service.

Reviewer: Same comments than for the backend image section

  - `registry`: The registry where the buildkit image is hosted. Defaults to `` (You can change all the registries by setting `globals.registry` to your registry).
  - `repository`: The repository path for the buildkit image. Defaults to `okteto/buildkit` if not specified.
  - `tag`: The image tag to deploy. Defaults to `{variables.chartVersion}` if not specified.
- `labels`: Labels to add to the buildkit pods.
- `podManagementPolicy`: The [podManagementPolicy](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#pod-management-policies) of the buildkit pods. Defaults to `Parallel`.
- `priorityClassName`: The priority class to be used by buildkit pods. The PriorityClass must already exist in your cluster before using this setting. This value has precedence over `globals.priorityClassName` if both are set. If this value is not set, the pods will inherit the priority class defined by the value set in `globals.priorityClassName`.

> - `rootless`: Specifies if the buildkit should be run rootless.

Reviewer: Could you document `rootless.enabled`?

  allowfullscreen
></iframe>

### cli

> The cli that will be used for performing tasks like starting synchronisation service, run inside your pipeline jobs, etc.

Reviewer: ```suggestion
Settings for the Okteto CLI.
```

> - `image`: Container image used by the cli operation.

Reviewer: ```suggestion
- `image`: Specifies the image settings for the Okteto CLI.
```

>   - `registry`: The registry where the cli image is hosted. Defaults to `` (You can change all the registries by setting `globals.registry` to your registry).

Reviewer: ```suggestion
  - `registry`: The registry where the Okteto CLI cli image is hosted. Defaults to `docker.io`.
```
Note that this doesn't default to `globals.registry`.

>   - `repository`: The repository path for the cli image. Defaults to `okteto/okteto` if not specified.

Reviewer: ```suggestion
  - `repository`: The repository path for the Okteto CLI image. Defaults to `okteto/okteto`.
```

- `extraEnv`: Environment variables to be set on the daemonset containers.
- `labels`: Labels to add to the daemonset pods.
- `image`: Container image used by the daemonset pods.
  - `registry`: The registry where the daemonset image is hosted. Defaults to `` (You can change all the registries by setting `globals.registry` to your registry).
  - `repository`: The repository path for the daemonset image. Defaults to `okteto/daemon` if not specified.

>   - `tag`: The image tag to deploy. Defaults to `0.14.0` if not specified.

Reviewer: Same here, I thought we were tagging all our images with the chart release on every release to simplify the airgap configuration

- `extraEnv`: Environment variables to be set on the daemonset containers.
- `labels`: Labels to add to the daemonset pods.
- `image`: Container image used by the daemonset pods.

>   - `registry`: The registry where the daemonset image is hosted. Defaults to `` (You can change all the registries by setting `globals.registry` to your registry).

Reviewer: same comments as the backend service

- `enabled`: Control if defaultBackend is deployed. Defaults to `true`.
- `extraEnvs`: Environment variables to be set on the defaultBackend containers.
- `image`: Configuration block to customize the image used by the defaultBackend pods.

>   - `registry`: registry for the defaultBackend pods (You can change all the registries by setting `globals.registry` to your registry)

Reviewer: same

- `annotations`: Annotations to add to the frontend pods.
- `extraEnv`: Environment variables to be set on the frontend containers.

> - `image`: Configuration block to customize the image used by the frontend pods.

Reviewer: same

- `extraEnv`: Environment variables to be set on the installer job containers.
- `labels`: Labels to add to the installer job pods.
- `runner`: to configure your [custom installer image](admin/custom-installer-image.mdx).

>   - `registry`: The registry where the installer image is hosted. Defaults to `` (You can change all the registries by setting `globals.registry` to your registry).

Reviewer 3: Is the default here supposed to be blank?

Reviewer: I would say "Defaults to the helm configuration setting `globals.registry`"

- `annotations`: Annotations to add to the installer job pods.
- `extraEnv`: Environment variables to be set on the installer job containers.
- `labels`: Labels to add to the installer job pods.

> - `image`: Specifies the image settings for your installer jobs.

Reviewer: The field is not `image`, it's `runner`

- `annotations`: Annotations to add to the registry pods.
- `extraEnv`: Environment variables to be set on the registry containers.
- `image`: Specifies the image settings for your registry service.

>   - `registry`: The registry where the registry image is hosted. Defaults to `` (You can change all the registries by setting `globals.registry` to your registry).

Reviewer: same

#### priorityClassName

> Defines the priority class to be used by all pods in the Okteto Namespace.

Reviewer: ```suggestion
Defines the priority class to be used by all pods in the `okteto` namespace. 
```
We use upper case for namespaces managed by Okteto, but we are referring to a Kubernetes namespace here

### injectDevelopmentBinaries

> Automatically inject kubectl, helm, and Okteto binaries on every development environment, and on the git and helm deployment pipelines. This requires permissions to mount a host volume.

Reviewer: ```suggestion
Automatically inject `kubectl`, `helm`, and `okteto` binaries on every development container, and on the pipeline runner.
```
We are referring to binaries here, not the Okteto platform
